tests/postgres_tests/test_serial.py

```diff
@@ -0,0 +1,344 @@
+        with connection.cursor() as cursor:
+            return connection.introspection.get_table_description(
+                cursor, model._meta.db_table
+            )
+
+
+class SerialModelTests(SerialModelBaseTests):
+    def test_create(self):
+        self.reset_sequences_to_1(SerialModel)
+
+        obj_1 = SerialModel.objects.create()
+        self.assertEqual(obj_1.small_serial, 1)
+        self.assertEqual(obj_1.serial, 1)
+        self.assertEqual(obj_1.big_serial, 1)
+
+        obj_2 = SerialModel.objects.create()
+        self.assertEqual(obj_2.small_serial, 2)
+        self.assertEqual(obj_2.serial, 2)
+        self.assertEqual(obj_2.big_serial, 2)
+
+        obj_3 = SerialModel.objects.create(
+            small_serial=None,
+            serial=None,
+            big_serial=None,
+        )
+        self.assertEqual(obj_3.small_serial, 3)
+        self.assertEqual(obj_3.serial, 3)
+        self.assertEqual(obj_3.big_serial, 3)
+
+        obj_4 = SerialModel.objects.create(
+            small_serial=111,
+            serial=222,
+            big_serial=333,
+        )
+        self.assertEqual(obj_4.small_serial, 111)
+        self.assertEqual(obj_4.serial, 222)
+        self.assertEqual(obj_4.big_serial, 333)
+
+        obj_5 = SerialModel.objects.create()
+        self.assertEqual(obj_5.small_serial, 4)
+        self.assertEqual(obj_5.serial, 4)
+        self.assertEqual(obj_5.big_serial, 4)
+
+        obj_6 = SerialModel.objects.create()
+        self.assertEqual(obj_6.small_serial, 5)
+        self.assertEqual(obj_6.serial, 5)
+        self.assertEqual(obj_6.big_serial, 5)
+
+        obj_7 = SerialModel.objects.create()
+        self.assertEqual(obj_7.small_serial, 6)
+        self.assertEqual(obj_7.serial, 6)
+        self.assertEqual(obj_7.big_serial, 6)
+
+        obj_8 = SerialModel.objects.create(
+            small_serial=1,
+            serial=1,
+            big_serial=1,
+        )
+        self.assertEqual(obj_8.small_serial, 1)
+        self.assertEqual(obj_8.serial, 1)
+        self.assertEqual(obj_8.big_serial, 1)
+
+        obj_9 = SerialModel.objects.create()
+        self.assertEqual(obj_9.small_serial, 7)
+        self.assertEqual(obj_9.serial, 7)
+        self.assertEqual(obj_9.big_serial, 7)
+
+    def test_full_clean_success(self):
+        small_lo, small_hi = (-32768, 32767)
+        lo, hi = (-2147483648, 2147483647)
+        big_lo, big_hi = (-9223372036854775808, 9223372036854775807)
+        test_cases = (
+            {},
+            {"small_serial": None},
+            {"serial": None},
+            {"big_serial": None},
+            {"small_serial": small_lo},
+            {"small_serial": small_hi},
+            {"serial": lo},
+            {"serial": hi},
+            {"big_serial": big_lo},
+            {"big_serial": big_hi},
+        )
+
+        for kwargs in test_cases:
+            with self.subTest(kwargs=kwargs):
+                SerialModel(**kwargs).full_clean()
```

Comment:
```suggestion
                self.assertIsNone(SerialModel(**kwargs).full_clean())
```
Optional

Reply:
`full_clean` never returns anything so I'm not sure what you're suggesting?

Reply:
`.full_clean()` will either raise an error or return None, so in some of the existing tests around `.full_clean()` it has an `assertIsNone` so that the test has an assert
It's not ideal to have tests without asserts, another option I've seen is saving afterwards and asserting the values

Reply:
Okay, if it's a matter of consistency then of course let's do it this way, I just personally disagree.